feelslikemonday/app/src/main/java/com/example/feelslikemonday/model/MoodEvent.java

```diff
@@ -0,0 +1,102 @@
+package com.example.feelslikemonday.model;
+
+import android.location.Location;
+
+//Todo: Do we give user's the ability to change time and date?
+// Also, should we have a list of all options stored as a static attribute?
+//The date and time are made as strings since it should be easier to perform operations to get them.
+//It doesn't seem we perform any operations that doesn't warrant them not to be strings
+public class MoodEvent {
+    private String date;
+    private String time;
+    private String emotionalState;
+    private String reason;
+    private String socialSituation;
+    private MoodType moodType;
+    private byte[] image;
+
+    //To be attached when editing
+    private Location location;
+
+    //Only one constructor since it's not possible to have overloading with different constructors
+    public MoodEvent(String date, String time, String emotionalState, String reason, MoodType moodType, String socialSituation) {
+        this.date = date;
+        this.time = time;
+        this.emotionalState = emotionalState;
+        this.reason = reason;
+        this.moodType = moodType;
+        this.socialSituation = socialSituation;
+
+    }
+
+    public MoodEvent(String date, String time, String emotionalState, String reason, MoodType moodType) {
```

Comment:
do these constructors cover all cases?
mandatory :  data, time, emotionalstate
optional:  reason, socialsituation, location, image 
I feel like not all cases are covered?

Reply:
Reason and socialsituation are accounted for. Made the assumption that location and image are added after the fact, and hence are not needed on object creation